reconcile/test/test_openshift_namespace_labels.py

```diff
@@ -0,0 +1,295 @@
+import json
```

Comment:
This test is very hard to follow in its current shape. It might be easier to understand if you separated it smaller pieces that contain the different scenarios which are easier to digest. If you need to prepare similar scenarios, the use of set up / tear down unit test functions to prepare the expected data that will be fed into a smaller unit test. A small comment about the general test strategy would be helpful.

The use of a unit testing framework such as unittest or pytest is definitely recommend while writing tests. It could also be helpful with your mocking strategy.

Reply:
The new tests look great. Things are much easier to understand, I love the usage of set up and tear down functions to avoid the repetition of annotations.

I agree, we don't need more tests for `reconcile()`